vsintegration/src/FSharp.ProjectSystem.FSharp/Project.fs

```diff
@@ -1323,6 +1323,26 @@ namespace rec Microsoft.VisualStudio.FSharp.ProjectSystem
             member x.GetCompileItems() = let sources,_ = sourcesAndFlags.Value in sources
             member x.GetCompileFlags() = let _,flags = sourcesAndFlags.Value in flags
 
+            /// Sync the source files with what's listed in the Solution Explorer.
+            /// This is called when a file is added, moved or removed from the project.
+            member this.SyncWithHierarchy() =
+                let sources =
+                    this.AllDescendants
+                    |> Seq.choose (function
+                        | :? FileNode as fileNode when fileNode.ItemNode.ItemName = ProjectFileConstants.Compile ->
```

Comment:
Does this include generated items produced by MSBuild tasks, e.g. pars.fs, lex.fs, assemblyinfo files, resources etc.?

Reply:
Can you give me an example project where I can test this?

Reply:
Microsoft.Compiler.Private.dll.

Or anything with generated attributes.

Reply:
Hmmm it's FSharp.Compiler.Private.dll of course, because we changed the name.

Reply:
Ah thank you - in which case no this doesn't work. I'll reimplement it to be a bit more clever 👍

Reply:
Super, thanks @saul, I'm glad I wasn't imagining something :)